app/controllers/my_module_shareable_links_controller.rb

```diff
@@ -4,23 +4,32 @@ class MyModuleShareableLinksController < ApplicationController
   before_action :load_my_module, except: %i(protocol_show
                                             repository_index_dt
                                             repository_snapshot_index_dt
-                                            download_asset)
+                                            download_asset
+                                            download_result_asset
+                                            results_show)
   before_action :check_view_permissions, only: :show
   before_action :check_manage_permissions, except: %i(protocol_show
                                                       repository_index_dt
                                                       repository_snapshot_index_dt
-                                                      download_asset)
+                                                      download_asset
+                                                      download_result_asset
+                                                      results_show)
   before_action :shareable_link_load_my_module, only: %i(protocol_show
                                                          repository_index_dt
                                                          repository_snapshot_index_dt
-                                                         download_asset)
+                                                         download_asset
+                                                         download_result_asset
+                                                         results_show)
   before_action :load_repository, only: :repository_index_dt
   before_action :load_repository_snapshot, only: :repository_snapshot_index_dt
   before_action :load_asset, only: :download_asset
+  before_action :load_result_asset, only: :download_result_asset
```

Comment:
Just load assets inside the actions.
And rename download_asset to download_step_asset